mcnic/aws-shop-cdk
# task-10

# Please take me time before deadline

# Task 10 (Backend For Frontend)

[LINK](https://github.com/rolling-scopes-school/aws/blob/main/aws-developer/10_backend_for_frontend/task.md)

1. Screenshot
![image](https://github.com/user-attachments/assets/81634c96-9ff3-4be5-a83f-3071782f2c2e)
![image](https://github.com/user-attachments/assets/c6e17875-b407-4992-a3e1-020ba5dde1a6)
![image](https://github.com/user-attachments/assets/4dbd744f-017c-4395-bbcd-279e247a2e82)

## Evaluation criteria (80 points for covering all criteria)

Provide your reviewers with the following information:
1. FE: <https://d3fihs3nsna7qy.cloudfront.net/>
1. link to the repo: <https://github.com/mcnic/aws-shop-cdk/bff-service>
1. Product Service API URL: <https://6jeal9ve0d.execute-api.us-east-1.amazonaws.com/prod>
      (sorry, this is some CORS problem)
        for testing need autorize, use header
        Authorization: `Basic bWNuaWM6VEVTVF9QQVNTV09SRA==`
1. Cart Service API URL: <http://mcnic-cart-api-prod3.us-west-2.elasticbeanstalk.com:4000/api>
1. BFF Service API URL: <http://mcnic-bff-api2.us-west-2.elasticbeanstalk.com:4000>
1. example of how to call createProduct lambda with all needed information:
      url `http://mcnic-bff-api2.eu-west-1.elasticbeanstalk.com:4000/api/products`
      method `POST`
      body ```{
          "title": "sample title",
          "description": "sample description",
          "price": 1,
          "image": "https://via.placeholder.com/150/000000/FFFFFF/?text=test",
          "count": 1
        }```
1. example how to call Product Service and Cart Service via BFF Service URL
  GET `http://mcnic-bff-api.eu-west-1.elasticbeanstalk.com/api/products`
  GET `http://mcnic-bff-api.eu-west-1.elasticbeanstalk.com/api/cart`

## Products representation on Home page should be based on Product Service API

- [x] Product images are not randomly generated on client side. Product image links, same as another product model information should be stored on BE side in Product DB.
- [x] Products might be created through CSV product file import from client side.

## Additional (optional) tasks

    How to test:
    1. Get products list
    1. Create new product
    1. Get products list - result shouldn’t have new product
    1. Wait more than 2 minutes
    1. Get products list - result should have new product

## Penalties

- [ ] 50 - Express package is used in JS/TS.

Done 14.8.2024

Total 48/100 points
